### DISCUSSION TITLE
How do i make wavy backgrounds on kade engine 1.8?

### DISCUSSION THREAD

**Discussion by BoagerEpic301** [KadeArchive/Kade-Engine#2881] [Q&A]
yeah

**Comment by BoloVEVO** [comment]
I recommend you using Kade Engine master branch (1.8.1 basically) cuz 1.8 is very buggy. Answering your question, you can use WiggleEffect.hx tools (in source folder in every FNF build) to make your wavy shaders. 

Go to Stage.hx 

There's an unused wiggleEffect shader in PlayState called "wiggleShit", so we're gonna use it B). In your custom stage set your wiggle effect parameters: 
Wiggle effect parameters example:
```
wiggleShit.effectType = WiggleEffectType.DREAMY; (Wave Style)
wiggleShit.waveAmplitude = 0.01; (Wave amplitude)
wiggleShit.waveFrequency = 60; (Wave Frequency)
wiggleShit.waveSpeed = 0.8; (Wave Speed)
```
If you have issues with this you can see how your shader looks with this haxe demo https://haxeflixel.com/demos/BlendModeShaders/

With your parameters now set, set your bg shader.
Example:
```
var mycoolbg:FlxSprite = new FlxSprite(x,y).loadGraphic(Paths.loadImage("urmom");
mycoolbg.shader = wiggleShit.shader;
```
And that's it.
